### PR TITLE
EVG-6948: project tags infrastructure

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: model/project_ref.go [hunk 1]
```diff
@@ -604,6 +605,9 @@ func (p *ProjectRef) AddTags(tags ...string) (bool, error) {
 		if _, ok := set[t]; ok {
 			continue
 		}
+		if strings.Contains(t, ",") {
```
Comment (bsamek) [inline model/project_ref.go]: Shouldn't this error?

Comment (tychoish) [reply model/project_ref.go]: done.

**Comment by bsamek** [comment]
@tychoish You've got some test and test compile failures.

File: model/project_ref.go [hunk 2]
```diff
@@ -207,6 +210,42 @@ func FindFirstProjectRef() (*ProjectRef, error) {
...
+		db.NoProjection,
+		db.NoSort,
+		db.NoSkip,
+		db.NoLimit,
+		&projectRefs,
+	)
+
+	if adb.ResultsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
```
Comment (bsamek) [inline model/project_ref.go]: nit: No need to put this in an else

**Pull request merged**